src/api/admin/types.ts

```diff
@@ -156,7 +156,43 @@ export type MembraneProof = Uint8Array;
 /**
  * @public
  */
-export type MemproofMap = { [key: string]: MembraneProof };
+export type MemproofMap = { [key: RoleName]: MembraneProof };
+
+/**
+ * @public
+ */
+export type RoleSettingsMap = { [key: RoleName]: RoleSettings };
+
+/**
+ * @public
+ */
+export type RoleSettings =
+  | {
+      type: "UseExisting";
+      cell_id: CellId;
+    }
+  | {
+      type: "Provisioned";
+      membrane_proof?: MembraneProof;
+      modifiers?: DnaModifiersOpt;
+    };
+
+/**
+ *  @public
+ * Any value that is serializable to a Yaml value
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Is this needed here? It should only produce a warning.
Actually `unknown` would be the idiomatic type.